[FIX] im_livechat: On top clickable element

nboulif opened the pull request:
Issue

	To do on Mobile:

	- Install "Live Chat" and "Ecommerce" modules
	- Go to "Live Chat" and join a channel.
	- Go to the shop on website
	- Add any product to basket and checkout

	On address form, impossible to click on next button.

Cause

	Chat button on top of "next" form button.

Solution

	Add close button to the "Live Chat" box to allow
	hiding it in case it's on top an element.

opw-2348624

nle-odoo reviewed (approved):
looks good, I'm not X button is necessary for desktop (there is bootstrap to hide something only when closed)

I don't see much how we could do it differently if we wanted to fix this.

couldn't `_closeChat` be used instead instead of _hideChat button?

And possibly you could just keep:

```js
 'click': '_openChat'
```

as is, and in:

```js
'click .o_livechat_hide': '_hideChat', // or _closeChat
```

I think you could do:

```js
_hideChat: function (e) {
    e.stopPropagation();
    // code content
}
```

this way everything but the X button still open the chat (which depending on style may not have been the case with `click .o_livechat_open`

nboulif commented:
Thanks for the review :slightly_smiling_face:  !

> looks good, I'm not sure X button is necessary for desktop (there is bootstrap CSS classes to show something only when in mobile), but on the other hand it would allow to have same experience.

Agreed. I updated (with `d-md-none`) to show only in mobile for now. 

> I don't see much how we could do it differently if we wanted to fix this.

I did it on the livechat module in case it can be usefull else where, but the main issue was in shop (like explained in ticket specs), and only with a specific theme (Graphene, not tested all others, but looks like OK for most). Maybe a fix can be done on this theme for this specific case, but not sure it's a good fix...

> couldn't _closeChat be used instead instead of _hideChat button?

_closeChat is used to destroy the chat window. In this case, the chat window is not opened yet so
I added the _hideChat to hide the $el...

> And possibly you could just keep:
>     'click': '_openChat'

Nice, I forgot the `e.stopPropagation();`, I just updated the _hideChat :+1:

nboulif commented:
@nle-odoo Thanks for the review :slightly_smiling_face:  !

> looks good, I'm not sure X button is necessary for desktop (there is bootstrap CSS classes to show something only when in mobile), but on the other hand it would allow to have same experience.

Agreed. I updated (with `d-md-none`) to show only in mobile for now. 

> I don't see much how we could do it differently if we wanted to fix this.

I did it on the livechat module in case it can be usefull else where, but the main issue was in shop (like explained in ticket specs), and only with a specific theme (Graphene, not tested all others, but looks like OK for most). Maybe a fix can be done on this theme for this specific case, but not sure it's a good fix...

> couldn't _closeChat be used instead instead of _hideChat button?

_closeChat is used to destroy the chat window. In this case, the chat window is not opened yet so
I added the _hideChat to hide the $el...

> And possibly you could just keep:
>     'click': '_openChat'

Nice, I forgot the `e.stopPropagation();`, I just updated the _hideChat :+1:

nle-odoo commented:
Hi,

I think after our discussion, better fix could be to have mb32 working in this given case:

https://github.com/odoo/odoo/blob/69cbd156b9f483024e89d0640de0fdd872fd41cf/addons/website_sale/views/templates.xml#L761-L762

So having something like:

```css
#wrapwrap > main > #wrap .oe_website_sale .oe_cart .btn.mb32 {
    margin-bottom: 32px!important; 
}
```

in theme_graphene and theme_avantgarde (that are the only ones it seems that rewrite `mb32`)

nboulif commented:
@nle-odoo Alright, thanks. I wanted to do it exactly like this but thought it's too "raw" :slightly_smiling_face:  Done :+1: 
https://github.com/odoo/design-themes/pull/403

nle-odoo reviewed (commented):
robodoo r+

nle-odoo commented:
fw-bot up to saas-13.4
